### ISSUE TITLE
ls()

### ISSUE THREAD

**Issue by ryansturmer** [ryansturmer/thinfat32#5]
It would be nice to have an ls() like function that would list the files in
a directory.  Maybe something with a static component so you could call it
over and over on a directory, and it would give you successive new paths
every time.

**Comment by fogleman** [comment]
static? no way man. think about thread safety.

**Comment by ryansturmer** [comment]
Embedded!  No threads!

Well, usually.  Either way, I guess a better way of putting it is "context retaining" rather than "static"

**Comment by kainjow** [comment]
I agree with @fogleman. If you're going to add this, just make a small struct as another input param for your context. That won't harm anything and makes thread safety a nice feature for those who want it.
